src/components/portal.ts

```diff
@@ -0,0 +1,24 @@
+import { addComponent, defineComponent, IWorld, Types } from 'bitecs';
+import { Position } from './position';
+import { Camera } from '../systems/renderer/camera';
+import { Radius } from './radius';
+
+export const Portal = defineComponent({
+    x: Types.f32,
+    y: Types.f32,
+});
+
+export function addTeleport(world: IWorld, camera: Camera, entity: number, from: number, to: number) {
+    addComponent(world, Portal, entity);
+
+    if (Position.x[entity] <= Position.x[from] + Radius.value[entity] &&
+        Position.x[entity] >= Position.x[from] - Radius.value[entity] &&
+        Position.y[entity] <= Position.y[from] + Radius.value[entity] &&
+        Position.y[entity] >= Position.y[from] - Radius.value[entity]
+    ) {
+        Position.x[entity] = Position.x[to];
+        Position.y[entity] = Position.y[to];
+        camera.x = Position.x[to];
+        camera.y = Position.y[to];
+    }
+}

```

Comment:
o componente apenas deve ter dados (comportamentos são nos sistemas)
o método de adição adiciona o componente portal então deve ser chamado de addPortal
o método de adição deve receber apenas as propriedades que será usada no componente


```suggestion
import { addComponent, defineComponent, IWorld, Types } from 'bitecs';

export const Portal = defineComponent({
    x: Types.f32,
    y: Types.f32,
});

export function addPortal(world: IWorld, entity: number, x: number, y: number) {
    addComponent(world, Portal, entity);
    Portal.x[entity] = x;
    Portal.y[entity] = y;
}

```

Reply:
capicce